tests/test_parser_stress.py

```diff
@@ -0,0 +1,350 @@
+        data = get_random_bytes(1024 * 1024, 259)
+        self.do_test(data, 'random_bytes', False)
+        pass
+
+    def test_special_chars_last_attach(self):
+        data = get_hyphens_crlfs(1024 * 1024, 359)
+        self.do_test(data, 'hyphens_crlfs', True)
+        pass
+
+    def test_special_chars_first_attach(self):
+        data = get_hyphens_crlfs(1024 * 1024, 459)
+        self.do_test(data, 'hyphens_crlfs', False)
+        pass
+
+    def do_test(self, data, data_name, last_part):
+
+        with BytesIO(data) as dataset_:
+            if last_part:
+                fields = {
+                    'name': 'hello world',
+                    'file': ('file.dat', dataset_, 'binary/octet-stream')
+                }
+            else:
+                fields = {
+                    'file': ('file.dat', dataset_, 'binary/octet-stream'),
+                    'name': 'hello world'
+                }
+
+            encoder = MultipartEncoder(fields=fields)
+            content_type = encoder.content_type
+            body = encoder.to_string()
+            encoder = None
+
+        interesting_numbers = get_interesting_numbers()
+        self.assertEqual(len(interesting_numbers), 880)
+
+        idx = 0
+        for default_chunksize in interesting_numbers:
+            idx += 1
+
+            with self.subTest(data_name=data_name,
+                              default_chunksize=default_chunksize):
+                # print(idx, '; name: ', data_name, '; last_part: ',
+                #       last_part, '; chunksize: ', default_chunksize)
+
+                parser = StreamingFormDataParser(
+                    headers={'Content-Type': content_type})
+
+                target = TestingValueTarget()
+                parser.register('file', target)
+
+                remaining = len(body)
+                offset = 0
+
+                while(remaining > 0):
+                    chunksize = min(remaining, default_chunksize)
+                    parser.data_received(body[offset:offset + chunksize])
+                    offset += chunksize
+                    remaining -= chunksize
+
+                self.assertEqual(offset, len(body))
+                self.assertEqual(target.multipart_filename, 'file.dat')
+                self.assertEqual(target.started, True)
+                self.assertEqual(target.finished, True)
+                result = target.value
+                self.assertEqual(len(result), len(data))
+                self.assertEqual(result, data)
+                result = None
+        self.assertEqual(idx, len(interesting_numbers))
+
+
+class DifferentFileSizeTestCase(TestCase):
+
+    def test_basic(self):
+        data = get_random_bytes(get_max_number(), 137)
+        self.do_test(data, 'random_bytes')
+        pass
+
+    def test_special_chars(self):
+        data = get_hyphens_crlfs(get_max_number(), 237)
+        self.do_test(data, 'hyphens_crlfs')
+        pass
+
+    def do_test(self, data, data_name):
```

Comment:
A small docstring (something like https://stackoverflow.com/a/3899154/179729) would be helpful here to understand what this function is doing. :)